### DISCUSSION TITLE
Updated dds version from 1.5.0 to 2.10.1, I got the memory reduced on vxworks platform

### DISCUSSION THREAD

**Discussion by zzhywzq** [eProsima/Fast-DDS#3980] [Q&A]
There is a demo based on cyberRT, the memory reduced 40M when I replaced the new dds library. How dose it take effect?

1. before: rtps1.5.0, memory 80M was used.
2. after: rtps2.10.1,memory 40M was used.

**Comment by JLBuenoLopez** [comment]
@zzhywzq 

There has been efforts to reduce the memory consumption of user's applications using Fast DDS. Nevertheless, it is so long since FastRTPS v1.5.0 has not been supported that I can hardly help you. After upgrading is everything working as you expected? If so, I would not be concerned about the memory reduction.

**Reply by zzhywzq** [reply]
It works fine, but it makes great sense to lower the memory when VxWorks system resource is limited.

**Comment by bjtucy** [comment]
Really interesting to see Fast DDS 2.10.1 running on VxWorks and reducing memory usage. I’m trying to build Fast DDS on VxWorks 6.6 with Diab 5.6.0.0, but I’m stuck because the compiler doesn’t fully support C++11 (especially std::array and move semantics).Could you share which VxWorks version and compiler you used? It would help me understand how you managed to build it. Thanks!
